nixos/lib/systemd-lib.nix

```diff
@@ -80,6 +80,10 @@ in rec {
     optional (attr ? ${name} && !elem attr.${name} values)
       "Systemd ${group} field `${name}' cannot have value `${toString attr.${name}}'.";
 
+  assertValuesAllOfOr = name: values: default:  group: attr:
```

Comment:
The name is a bit misleading, what about `assertValuesSomeOfOr`?

Reply:
I agree, that fits much better. I'll adjust it directly in the appropriate commit so as not to mess up the history.

Reply:
Sorry for the inconvenience, I also adjusted the 2nd commit to only allow certain values by using `listOf enum [...]` as the type for the option to detect misconfigurations early.